src/index.ts

```diff
@@ -0,0 +1,130 @@
+export type Networks = 'rinkeby' | 'mainnet';
+
+export interface SdkInstance {
+  addListeners: (listeners: SafeListeners) => void;
+  removeListeners: () => void;
+  sendTransactions: (txs: any[]) => void;
+}
+
+export interface SafeInfo {
+  safeAddress: string;
+  network: Networks;
+  ethBalance: string;
+}
+
+export interface SafeListeners {
+  onSafeInfo: (info: SafeInfo) => any;
+}
+
+export enum FromSafeMessages {
+  ON_SAFE_INFO = 'ON_SAFE_INFO',
+}
+
+export enum ToSafeMessages {
+  SEND_TRANSACTIONS = 'SEND_TRANSACTIONS',
+}
+
+const config: {
+  safeAppUrl?: string;
+  listeners?: SafeListeners;
+} = {};
+
+const _logMessageFromSafe = (origin: string, message: FromSafeMessages) => {
+  console.info(`SafeConnector: A message with id ${message} was received from origin ${origin}.`);
+};
+
+const _onParentMessage = async ({ origin, data }: MessageEvent) => {
+  if (origin === window.origin) {
+    return;
+  }
+
+  if (origin !== config.safeAppUrl) {
```

Comment:
Can we allow this to be an array or pattern (so that you could allow all different envs), maybe even make it optional/ nullable

Note: not super imporant in the first version.

Reply:
the only exposed method is initSdk, there you have to pass the URL. you will use the safe app url depending on your current environment. So you will have to handle that outside the lib.

for example: `initSdk(proccess.env.safeAppUrl)`

Reply:
What I meant is, that I have a Safe app and it should not be specific to 1 safe url. I am not sure if it would be easier if I could specify the interface I support (e.g. the production rinkeby and mainnet version), rather than having to somehow pull the iframe owner (not sure how that works in js)

Reply:
I think we are talking about the same. I think there is not need to handle that in the lib.

for example...
```
const safeUrl = network === 'mainnet' ? 'https:safeMainnet.com' : 'http://localhost:3002'
sdkInit(safeUrl)
```
If you think this can limit us in somehow, we can implement the array of URLs